packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,158 @@
+import { Component, Element, h, Host, Prop, Watch } from '@stencil/core';
+
+import {
+  PROGRESS_BORDER_SHAPE,
+  PROGRESS_THICKNESS,
+  PROGRESS_TYPE,
+  TProgressBorderShape,
+  TProgressThickness,
+  TProgressType,
+} from './bq-progress.types';
+import { validatePropValue } from '../../shared/utils';
+
+@Component({
+  tag: 'bq-progress',
+  styleUrl: './scss/bq-progress.scss',
+  shadow: true,
+})
+export class BqProgress {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqProgressElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** If `true` the indeterminate state of progress bar is enabled */
+  @Prop({ reflect: true }) indeterminate: boolean = false;
+
+  /** A number representing the current value of the progress bar */
+  @Prop({ reflect: true }) value = 0;
```

Comment:
You need to set this property as mutable since it can change its value internally in the `validateValue` method.

```suggestion
  @Prop({ reflect: true, mutable: true }) value = 0;
```